pkg/danm/danm.go

```diff
@@ -183,13 +191,115 @@ func extractConnections(args *cniArgs) error {
   return nil
 }
 
+func getResourcePrefix(args *cniArgs, resourceType string)(string,error){
+  confArgs, err := loadNetConf(args.stdIn)
+  if err != nil {
+    return "", errors.New("cannot load CNI NetConf due to error:" + err.Error())
+  }
+  k8sClient, err := createK8sClient(confArgs.Kubeconfig)
+  if err != nil {
+    return "", errors.New("cannot create K8s REST client due to error:" + err.Error())
+  }
+  configmap, err := k8sClient.CoreV1().ConfigMaps(string("kube-system")).Get(string("resource-prefix-map"), meta_v1.GetOptions{})
+  if err != nil {
+    return "", errors.New("failed to get 'resource-prefix-map' Config Map data from K8s API server due to:" + err.Error())
+  }
+  return  configmap.Data[resourceType], nil
+}
+
+func getRegisteredDevices(args *cniArgs, cniType string)([]string,error){
+  resourceMap := make(map[string]*checkpoint_types.ResourceInfo)
+  if string(args.podUid) != "" {
```

Comment:
can this ever happen? IMO Pod UUID is a mandatory param according to the CNI spec, but pls double check

if yes,remove the unnecessary complexity from the code

Reply:
Code has been refactored